tests/test_updater_with_simulator.py

```diff
@@ -257,6 +257,42 @@ def test_not_loading_targets_twice(self, wrapped_open: MagicMock) -> None:
         updater.get_targetinfo("somepath")
         wrapped_open.assert_not_called()
 
+    def test_update_metadata_to_cache(self) -> None:
+        updater = self._run_refresh()
+
+        # Add targets to repository
+        self.sim.targets.version += 1
+        self.sim.add_target("targets", b'some content', self.targets_dir)
+        self.sim.update_snapshot()
+
+        with patch.object(
+                self.sim, "_fetch_metadata", wraps=self.sim._fetch_metadata
+            ) as wrapped_fetch:
+
+            # Make a successful update of valid metadata which stores it in cache
+            updater = self._run_refresh()
+
+            # Clean up calls to fetch during refresh()
+            wrapped_fetch.reset_mock()
+
+            updater.get_targetinfo(self.targets_dir)
+
+            # Create a new updater and perform a second update while
+            # the metadata is already stored in cache (metadata dir)
+            updater = Updater(
+                self.metadata_dir,
+                "https://example.com/metadata/",
+                self.targets_dir,
+                "https://example.com/targets/",
+                self.sim,
+            )
+            updater.get_targetinfo(self.targets_dir)
+
+            # Test that snaphot and targets are loaded from cache and not downloaded
+            # and that timestamp is loaded from cache and used to verify the new timestamp
+            wrapped_fetch.assert_has_calls([call("root", 2), call("timestamp", None)])
```

Comment:
testing this has some merit... but I bet we could add this test to `test_top_level_roles_update()`and `test_delegated_roles_update()`:  existing tests that already test whether the expected files are being fetched.

However, would like to _actually verify what we load from disk_. The PR currently does not do that. So my (possibly naive) ideas was to patch the builtin open() and see what gets loaded from disk during `Updater()`, `refresh()` and `get_targetinfo()`